src/services/websockets.js

```diff
@@ -20,26 +20,32 @@ export function initialise (actionCreatorBinder) {
   } = actionCreatorBinder({...SearchResultActions, ...TagActions});
   primus.on('data', function received (data) {
     console.log('incoming socket data', data);
-    if (data.connection) {
-      saveSocketConnectionId(data.connection);
+    if (data.graphql) {
+      saveSearchResult(data);
+    }
+  });
+
+  primus.once('open', (data) => {
```

Comment:
Yes, the re-connection is automatic, and triggers the `reconnected` event at https://github.com/numo-labs/isearch-ui/pull/182/files#diff-e5dc273ff838fcfd0afb925ff66c6e53R32 which re-subscribes to the pre-existing primus room.